### PR TITLE
docs: add interactivity to the Folder Structure page

### LINKED ISSUE #810: docs: Add Interactivity (collapse/filter) to "Folder Structure" page
### Is your feature request related to a problem? Please describe.

Currently, the "Folder Structure" page in the docs includes a large text-based directory listing as well as several sub-headings which in many cases include `<sub>` tags to denote when the described item will be present (based on which components of the T3 stack are present). This leads to a lot of scrolling, and some visual confusion between h3/h4 headers (which are currently used to denote child directories).

### Describe the solution you'd like to see

I believe adding some interactivity to this page, where a user could configure/select which components they are using, and the page would respond to the selections, only showing relevant directory listing items as a result. Initially I think having a few buttons at the top w/ "[un]*selected" states could work quite well, as long as the toggle is obvious and it's easy to clear the selection. At minimum I believe the textual descriptions lower in the page should show/hide themselves automatically, as this feature alone would remove the need for the 22 `<sub>` tags that are currently in the page.

### Describe alternate solutions

Additionally, the directory listing itself could respond to the user configuration such that you don't see certain items...I believe this could be more disorienting/confusing for a user learning the framework however, so a better option might be to de-emphasize the unused portions on select, e.g. change font color and add italics, or add a strikethrough (when a component is not being used based on their selection).

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ronanru**
Closes #810

The guy who created the issue wanted to add this feature himself, but the issue was inactive for 3 weeks, so I implemented it.

## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog

- Converted the Folder Structure page to the MDX
- Added a form to select components of the T3 stack (~~Uses URLSearchParams to store the selection~~)
- Added a diagram component that generates the folder structure diagram for each configuration
- Added a script to hide the docs for files that aren't used in the selected configuration

---

## Screenshots

image

💯

### CODE REVIEW

**Review by c-ehrlich — COMMENTED** [review]
Seems to work as described :)

Haven't read the code in depth yet but personally I think the experience would be nicer if it:
- Updates without having to click a button 
- Defaults to everything on

But curious what others think

**Comment by juliusmarminge** [comment]
> Seems to work as described :)
> 
> Haven't read the code in depth yet but personally I think the experience would be nicer if it:
> 
> * Updates without having to click a button
> * Defaults to everything on
> 
> But curious what others think

Yea those sounds like good defaults

**Comment by ronanru** [comment]
> I think the experience would be nicer if it:
> 
> * Updates without having to click a button
> * Defaults to everything on

Thank you for the suggestions, I implemented these changes.

File: www/src/pages/en/folder-structure.mdx [hunk 1]
```diff
@@ -5,224 +5,233 @@ layout: ../../layouts/docs.astro
...
 ---
 
+import Form from "../../components/docs/folderStructureForm.astro";
+import Diagram from "../../components/docs/folderStructureDiagram.astro";
+
+Please select your packages to see the folder structure of a newly scaffolded app with those selections.
+
+<Form />
+
 The following is the folder structure of a newly scaffolded T3 App, with all options selected.
```
Comment (c-ehrlich) [inline www/src/pages/en/folder-structure.mdx:15]: This sentence needs to be changed now as it's showing the folders for the selected options, not for all options

**Review by c-ehrlich — DISMISSED** [review]
Just one more thing I noticed

File: www/src/pages/en/folder-structure.mdx (hunk 1 above)
Comment (juliusmarminge) [inline www/src/pages/en/folder-structure.mdx:15]:
```suggestion
```

just remove it?

**Review by juliusmarminge — APPROVED** [review]
Awesome feature :)

**Comment by juliusmarminge** [comment]
We might want to hide elements from the TOC that's not actually displayed:

CleanShot 2022-12-15 at 23 44 55

**Pull request merged**